### PR TITLE
feat:新增表单评分组件

### PR DESCRIPTION
**Pull request by 867-Jige**
## 变更点(Changes)

- xxxx

## 相关issues (Which issues this PR fixes)

- Fixes #

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-lesscode?pullRequest=794) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/bk-lesscode?pullRequest=794) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/bk-lesscode?pullRequest=794) it.</sub>

File: lib/client/src/components/render-nocode/form/components/form-edit/association-value/relation-rules.vue [hunk 1]
```diff
@@ -273,7 +295,7 @@
                 target.value = val
                 this.change()
             },
-            change () {
+            change (e) {
```
Comment (luofann) [inline lib/client/src/components/render-nocode/form/components/form-edit/association-value/relation-rules.vue:298]: 没有用到参数`e`

File: lib/client/src/components/render-nocode/form/components/form-edit/default-value.vue [hunk 2]
```diff
@@ -81,6 +82,7 @@
         },
         data () {
             return {
+
```
Comment (luofann) [inline lib/client/src/components/render-nocode/form/components/form-edit/default-value.vue:85]: 多余的空行

File: lib/client/src/components/render-nocode/form/components/form-edit/default-value.vue [hunk 3]
```diff
@@ -106,6 +108,7 @@
                 }
                 return dftVal
             },
+
```
Comment (luofann) [inline lib/client/src/components/render-nocode/form/components/form-edit/default-value.vue:111]: 多余的空行

File: lib/client/src/views/edit-nocode/components/more-actions/data-manage-operate.vue [hunk 4]
```diff
@@ -50,6 +50,7 @@
         methods: {
             // 获取表单数据管理页关联的表单页
             async getFormPage () {
+                // console.log(this.pageDetail)
```
Comment (luofann) [inline lib/client/src/views/edit-nocode/components/more-actions/data-manage-operate.vue:53]: 去掉调试语句

File: lib/client/src/views/edit-nocode/index.vue [hunk 5]
```diff
@@ -131,14 +132,13 @@
...
                     ])
-
                     await this.$store.dispatch('variable/getAllVariable', {
                         projectId: this.projectId,
                         pageCode: pageDetail.pageCode,
                         versionId: this.versionId,
                         effectiveRange: 0
                     })
-
+                    // console.log(pageDetail)
```
Comment (luofann) [inline lib/client/src/views/edit-nocode/index.vue:141]: 去掉调试语句

File: lib/client/src/views/project/index.vue [hunk 6]
```diff
@@ -62,16 +49,7 @@
...
-                                <!-- <span>{{ childrenItem.title }}</span> -->
                                 <template v-if="childrenItem.iamAction">
-                                    <!-- <auth-component
-                                        :permission="childrenItem.iamAction === 'develop_app' ? curProject.canDevelop : curProject.canDeploy"
-                                        :auth="childrenItem.iamAction"
-                                        :resource-id="$route.params.projectId"
-                                        @before-show-permission-dialog="beforeShowPermissionDialog">
-                                        <a href="javascript:;" slot="forbid" custom-forbid-container-cls="menu-child-forbid-container-cls">{{childrenItem.title}}</a>
-                                        <span slot="allow">{{childrenItem.title}}</span>
-                                    </auth-component> -->
```
Comment (luofann) [inline lib/client/src/views/project/index.vue:74]: 上面这些注释的代码，和本次功能无关的不需要删掉。其他开发者保留可能会有原因。

File: lib/client/src/components/flow-form-comp/form/index.vue [hunk 7]
```diff
@@ -167,18 +187,25 @@
...
+                        if (fieldType === 'RATE') {
+                            return this.getRateSatisfyIntervalRule(group, relation)
                         } else {
-                            return value in this.localValue && isEqual(this.localValue[relFieldKey], this.localValue[value])
+                            if (!(relFieldKey in this.localValue)) {
+                                return
+                            }
+                            if (type === 'CONST') {
+                                return isEqual(this.localValue[relFieldKey], value)
+                            } else {
```
Comment (luofann) [inline lib/client/src/components/flow-form-comp/form/index.vue:206]: 不需要写`else`了

File: lib/client/src/components/flow-form-comp/form/index.vue [hunk 8]
```diff
@@ -136,10 +142,24 @@
...
+                                    if ((item.min <= triggerValue) && triggerValue < item.max) {
+                                        value = item.value
+                                    }
+                                })
+                            } else {
+                                value = rule.target.type === 'CONST' ? rule.target.value : this.localValue[rule.target.value]
+                            }
                             this.localValue[field.key] = value
+                        } else if (field.type === 'RATE') {
+                            this.localValue[field.key] = 0
```
Comment (luofann) [inline lib/client/src/components/flow-form-comp/form/index.vue:162]: 没有找到满足条件的规则时，会把评分的值置为0吗？

File: lib/client/src/components/render-nocode/form/components/form-edit/association-value/relation-rules.vue [hunk 9]
```diff
@@ -178,7 +188,10 @@
                 if (['INT', 'DATE', 'DATETIME', 'MEMBER', 'MEMBERS'].includes(type)) {
                     return fields.filter(item => item.type === type)
                 } else if (!MULTIPLE_VALUE_TYPES.includes(type)) {
-                    return fields.filter(item => COMPARABLE_VALUE_TYPES.includes(item.type) && !MULTIPLE_VALUE_TYPES.includes(item.type))
+                    return fields.filter(item => {
+                        const bool = COMPARABLE_VALUE_TYPES.includes(item.type) && !MULTIPLE_VALUE_TYPES.includes(item.type)
+                        return type === 'RATE' ? item.type === 'RATE' && bool : bool
+                    })
```
Comment (luofann) [inline lib/client/src/components/render-nocode/form/components/form-edit/association-value/relation-rules.vue:194]: 评分组件只能引用相同类型变量的值，在上面的if判断语句增加`RATE`就可以了

File: lib/client/src/components/render-nocode/form/components/form-edit/association-value/relation-rules.vue [hunk 10]
```diff
@@ -194,7 +207,16 @@
             },
             // 规则字段可选列表
             getRelFieldList () {
-                return this.isCurrentTable ? this.fieldsList.filter(item => item.key !== this.field.key) : this.otherTableFields
+                const relFieldList = this.isCurrentTable ? this.fieldsList.filter(item => {
+                    let optionBool = item.key !== this.field.key
+                    // 评分组件只能选择数字类型的字段
```
Comment (luofann) [inline lib/client/src/components/render-nocode/form/components/form-edit/association-value/relation-rules.vue:212]: 规则条件字段不受类型的影响

**Review by luofann — CHANGES_REQUESTED** [review]

File: lib/client/src/components/flow-form-comp/form/fields/rate.vue [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    </div>
+</template>
+
+<script>
+    export default {
+        name: 'Rate',
+        props: {
+            value: {
+                type: [Number, String],
+                default: '{}'
```
[the remark this answers is not in the document]
Comment (867-Jige) [reply lib/client/src/components/flow-form-comp/form/fields/rate.vue:13]: 在lib\client\src\components\flow-form-comp\form\fieldItem.vue文件中的传入值，有可能是‘’
value: {
                type: [String, Number, Boolean, Array],
                default () {
                    return deepClone(FIELDS_TYPES.find(item => item.type === this.field.type)?.default || '')
                }
            },

**Review by terlinhe — APPROVED** [review]

**Pull request merged**